DAL/EmployeeDAL.cs

```diff
@@ -0,0 +1,218 @@
+        employeeToUpdate.Dob = updatedEmployee.Dob == DateTime.MinValue ? null :
+                            updatedEmployee.Dob ?? employeeToUpdate.Dob;
+        employeeToUpdate.Email = updatedEmployee.Email == "--d" ? null :
+                                 string.IsNullOrWhiteSpace(updatedEmployee.Email) ? employeeToUpdate.Email :
+                                 updatedEmployee.Email;
+        employeeToUpdate.MobileNumber = updatedEmployee.MobileNumber == "--d" ? null :
+                                         string.IsNullOrWhiteSpace(updatedEmployee.MobileNumber) ? employeeToUpdate.MobileNumber :
+                                         updatedEmployee.MobileNumber;
+        employeeToUpdate.JoiningDate = updatedEmployee.JoiningDate == DateTime.MinValue ? null :
+                                        updatedEmployee.JoiningDate ?? updatedEmployee.JoiningDate;
+        employeeToUpdate.LocationId = updatedEmployee.LocationId == -1 ? null :
+                                   updatedEmployee.LocationId ?? employeeToUpdate.LocationId;
+        employeeToUpdate.JobTitle = updatedEmployee.JobTitle == "--d" ? null :
+                                    string.IsNullOrWhiteSpace(updatedEmployee.JobTitle) ? employeeToUpdate.JobTitle :
+                                    updatedEmployee.JobTitle;
+        employeeToUpdate.DepartmentId = updatedEmployee.DepartmentId == -1 ? null :
+                                     updatedEmployee.DepartmentId ?? employeeToUpdate.DepartmentId;
+        employeeToUpdate.AssignManager = updatedEmployee.AssignManager == "--d" ? null :
+                                          string.IsNullOrWhiteSpace(updatedEmployee.AssignManager) ? employeeToUpdate.AssignManager :
+                                          updatedEmployee.AssignManager;
+        employeeToUpdate.AssignProject = updatedEmployee.AssignProject == "--d" ? null :
+                                          string.IsNullOrWhiteSpace(updatedEmployee.AssignProject) ? employeeToUpdate.AssignProject :
+                                          updatedEmployee.AssignProject;
+
+        int index = existingEmployees.FindIndex(emp => emp.EmpNo == empNo);
+        existingEmployees[index] = employeeToUpdate;
+        jsonUtils.WriteJSON(existingEmployees, filePath);
+        return true;
+    }
+
+    public bool Delete(string empNo)
+    {
+        List<Employee> existingEmployees = jsonUtils.ReadJSON<Employee>(filePath);
+        Employee employeeToDelete = existingEmployees.FirstOrDefault(employee => employee.EmpNo == empNo);
+        if (employeeToDelete != null)
+        {
+            existingEmployees.Remove(employeeToDelete);
+            jsonUtils.WriteJSON(existingEmployees, filePath);
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    private static EmployeeDetails ConvertToEmployeeDetails(Employee employee)
+    {
+
+        EmployeeDetails employeeDetails = new()
```

Comment:
you can have another customer in EmployeeDetails itself right with Employee as parameter